src/flow_solver/flow_solver.cpp

```diff
@@ -425,10 +427,15 @@ int FlowSolver<dim,nstate>::run() const
         // Time advancement loop with on-the-fly post-processing
         //----------------------------------------------------
         pcout << "Advancing solution in time... " << std::endl;
+        dealii::Timer timer(this->mpi_communicator,false);
+        timer.start();
         while((ode_solver->current_time) < (final_time - 1E-13)) //comparing to 1E-13 to avoid taking an extra timestep
         {
             // update adaptive time step
-            if(flow_solver_param.adaptive_time_step == true) time_step = flow_solver_case->get_adaptive_time_step(dg);
+            if(flow_solver_param.adaptive_time_step == true) {
+                time_step = flow_solver_case->get_adaptive_time_step(dg);
+                flow_solver_case->set_time_step(time_step);
+            }
 
             // advance solution
             ode_solver->step_in_time(time_step,false); // pseudotime==false
```

Comment:
I thought it would be more expensive to query from the `flow_solver_case` every time; I had only added this to access the time step within the flow_solver_case for knowing when to output the velocity field files

Reply:
Sounds good.